crypto/fipsmodule/rand/snapsafe_detect.c

```diff
@@ -0,0 +1,140 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/crypto.h>
+
+#include "snapsafe_detect.h"
+
+#if defined(OPENSSL_LINUX)
+#include <fcntl.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+#include <unistd.h>
+#include "../delocate.h"
+
+/* Snapsafety state */
+#define SNAPSAFETY_STATE_FAILED_INITIALISE 0x00
+#define SNAPSAFETY_STATE_SUCCESS_INITIALISE 0x01
+#define SNAPSAFETY_STATE_NOT_SUPPORTED 0x02
+
+DEFINE_STATIC_ONCE(aws_snapsafe_init)
+DEFINE_BSS_GET(const char *, sgc_file_path)
```

Comment:
Does this need to be a variable? Can the two places that use the file path use AWSLC_SYSGENID_PATH directly?

Reply:
Good point.  There's no need for a variable.